server/game/cards/08.1-TAK/MagTheMighty.js

```diff
@@ -7,28 +8,32 @@ class MagTheMighty extends DrawCard {
                 afterChallenge: event => event.challenge.winner === this.controller && this.isParticipating()
             },
             target: {
-                cardCondition: card => card.location === 'play area' && card.controller === this.controller && card.getType() === 'character'
+                cardCondition: card => card.location === 'play area' && card.controller === this.controller && card.getType() === 'character',
+                gameAction: 'kill'
             },
+            message: '{player} uses {source} to kill {target}',
             handler: context => {
-                this.game.killCharacter(context.target);
-                this.game.addMessage('{0} is forced by {1} to kill {2}', context.player, this, context.target);
-
-                //TODO Technically should only trigger when the first kill was not saved
-                this.game.promptForSelect(context.event.challenge.loser, {
-                    activePromptTitle: 'Select a character (only when first kill was not saved)',
-                    cardCondition: card => card.location === 'play area' && card.controller === context.event.challenge.loser && card.getType() === 'character',
-                    source: this,
-                    onSelect: (player, card) => this.onCardSelected(player, card)
-                });
+                this.game.resolveGameAction(
+                    GameActions.kill(context => ({
+                        card: context.target
+                    })).then(preThenContext => ({
+                        target: {
+                            cardCondition: { location: 'play area', type: 'character', controller: preThenContext.event.challenge.loser },
+                            gameAction: 'kill'
+                        },
```

Comment:
The choosing of the target should be done by the losing opponent. Currently, the winner/controller of mag chooses the opponent´s character to kill. I think adding `choosingPlayer` to the `target` should work, but i am not sure if you can use `preThenContext.event.challenge.loser` as the value for `choosingPlayer`

Reply:
Good catch, thanks!